### DISCUSSION TITLE
Questions about Bladed-style dll interface

### DISCUSSION THREAD

**Discussion by venturi123** [OpenFAST/openfast#1302] [Q&A]
Hi all,

I have a few questions about the control system in OpenFAST.

(1) I notice that the controller in the [reference](https://github.com/OpenFAST/r-test/tree/a8b4237a02aabc0822db7ea3681aff81c8b8947d/glue-codes/fast-farm/TSinflow) is about pitch and torque. I wonder if it's possible to control rotor speed directly according to the DLL function. Because I only care about the flow field rather than the response of the wind turbine structure, A simple control of pitch and rotor speed according to a given look-up table can achieve this goal. 

(2) I find that the results of rotor speed (avrSWAP(21) in Bladed-style DLL) and generator speed (avrSWAP(20) in Bladed-style DLL) are not equal when GBRatio=1 and RotSpeed=18.0. The ratio of the Generator speed to rotor speed keeps 9.5492, which is equivalent to the factor to convert radians per second to revolutions per minute, shown in figure 1. So, I wonder if there is something wrong with the unit of these two variables because both are defined in rad/s according to the user manual, shown in figure 2.

![image](https://user-images.githubusercontent.com/19996167/197929900-39f8d77d-e2af-4d96-b4c5-f245a5707633.png)
figure 1

![image](https://user-images.githubusercontent.com/19996167/197929751-49ffdf68-7e0b-48e2-af94-841735fa7b99.png)
figure 2

**Comment by venturi123** [comment]
Sorry for my carelessness. I found the reason for Q(2). The RotSpeed is changed through a self-defined function. 
BTW, I would very much appreciate it if someone could give me any suggestions about Q(1) : )

**Reply by jjonkman** [reply]
Dear @venturi123,

Unless you want to model a fixed generator speed, which can be achieved by disabling the generator degree of freedom (`GenDOF` = False), the generator azimuth is a DOF of the ElastoDyn module, the motion of which cannot be specified directly without reformulation of the equations of motion.  Rather you must implement some form of control (torque, pitch, etc.) through the ServoDyn module of OpenFAST to regulate the generator speed, just as you would in a real wind turbine.

Blade pitch is not currently a degree of freedom, and so, can be specified directly (without active control).

Best regards,
